### PR TITLE
school library: Add UI

### PR DESCRIPTION
**Pull request by manzitresor**
Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
```
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.
```

- [ ] In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

```
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.
```

### CODE REVIEW

File: main [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+def list_persons(app)
+  app.list_all_people
+  main(app)
+end
+
+def create_rental(app)
+  app.create_rental
+  main(app)
+end
```
Comment (Owusu-Desmond) [inline main]: - [ ] Please from the project requirement, all these methods should be in the `app.rb` file which serves as your console app entry-point, and this `main.rb` should only define the entry point by presenting the user with a list of options to perform, It lets users choose an option, if needed, ask for parameters for the option and have a way to quit the app like example code below. :relaxed: 
    
    ```ruby
    # main.rb
    def main
      app = App.new()
    
      response = nil
    
      # [OPTIONAL] You can learn  about the ruby lessons about basic sintax like loops: https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/lessons/basic_syntax.md  
      # *Present the user with a list of options to perform.
      puts "Welcome to School Library App!\n\n"
      while response != "7"
        puts "Please choose an option by enterin a number:"
        puts "1 - List all books"
        puts "2 - List all people"
        puts "3 - Create a person"
        puts "4 - Create a book"
        puts "5 - Create a rental"
        puts "6 - List all rentals for a given person id"
        puts "7 - Exit"
        # *Lets users choose an option.
        response = gets.chomp
    
        case response
        when "1"
          app.list_books()
        when "2"
          app.list_people()
        when "3"
          app.create_person()
        when "4"
          app.create_book()
        when "5"
          app.create_rental()
        when "6"
          app.list_rentals_for_person_id()
        when "7"
          # *Have a way to quit the app.
          puts "Thank you for using this app!"
        end
    
        puts "\n"
      end
    end
    
    # 'main' method should be invoked at the end of the main.rb file, hint in this lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point
    main()
    ```

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    person_index = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def get_rental(id)
+    person = people.find { |person_id| person_id.id == id }
+    puts 'Rentals:'
```
Comment (Owusu-Desmond) [inline app.rb]: - [ ] I received an error while attempting to list rentals by a person ID. To resolve this, I recommend updating the code from `person = people.find { |person_id| person_id.id == id }` to `person = @people.find { |person_id| person_id.id == id }`. This change should resolve the issue. Thank you.

  
  Screenshot from 2023-08-30 22-18-19

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @manzitresor , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    when 3
+      create_person_option
+    when 4
+      create_book_option
+    when 5
+      create_rental
+    when 6
+      rental_option
+    end
+  end
```
Comment (danmainah) [inline app.rb]: - [ ] @manzitresor Great work implementing the change requested, as the previous reviewer had requested kindly consider moving this section to your main.rb file as required in the project description.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @manzitresor , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implemented the change requested to ensure the user is able to view rentals.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Strangeal — APPROVED** [review]
Hi @manzitresor, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
-  Added requested changes ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**